apps/o2-blocks/src/p2-autocomplete/editor.js

```diff
@@ -10,29 +10,27 @@ import { map, unescape } from 'lodash';
  */
 import './editor.scss';
 
-/*
- * This is a workaround for the way Gutenberg autocomplete works. It only looks for
- * beginnings of words and something like "teamabcp2" won't be found if you search "abc".
- * Adding spaces around the word solves the problem.
- */
-const COMMON_PREFIXES = /(team|a8c|woo|happiness)/i;
-const stripCommonWords = str => str.replace( COMMON_PREFIXES, ' ' );
+function substrings( name, subs = [ name ] ) {
+	if ( name.length === 0 ) {
```

Comment:
Maybe don't include one character matches?

```suggestion
	if ( name.length <= 1 ) {
```

Reply:
Just for what it's worth I don't agree with the premise that `ty` would be wrong. I can't speak for others but personally I rely on two-character abbreviations all the time. Development-wise I'd be in favor of merging as-is and then seeing if this were a problem. Ultimately it won't matter much either way: if you type `ty` and fail to find `tyrian` then the third letter, the `r` will wipe out the false positives with almost no effort or attention.

If we had a function in Gutenberg we could hook into to weigh the results then we could weigh the results showing `serenity` for `ty` but prioritizing `tyrian` since it starts with `ty`. This is a reasonable use-case for such a function, as is the search-by-title issue.